sockeye/data_io.py

```diff
@@ -495,211 +1215,142 @@ def __init__(self,
-            buff_source[:source_len] = source
-            buff_target[:target_len] = target
-            buff_label[:len(target)] = target[1:] + [self.eos_id]
-            self.data_source[buck_idx].append(buff_source)
-            self.data_target[buck_idx].append(buff_target)
-            self.data_label[buck_idx].append(buff_label)
-            self.data_target_average_len[buck_idx] += target_len
-
-        # Average number of non-padding elements in target sequence per bucket
-        for buck_idx, buck in enumerate(self.buckets):
-            # Case of empty bucket -> use default padded length
-            if self.data_target_average_len[buck_idx] == 0:
-                self.data_target_average_len[buck_idx] = buck[1]
-            else:
-                self.data_target_average_len[buck_idx] /= len(self.data_target[buck_idx])
-
-        # We now have sufficient information to populate bucket batch sizes
-        self._populate_bucket_batch_sizes()
-
-        logger.info("Source words: %d", tokens_source)
-        logger.info("Target words: %d", tokens_target)
-        logger.info("Vocab coverage source: %.0f%%", (1 - num_of_unks_source / tokens_source) * 100)
-        logger.info("Vocab coverage target: %.0f%%", (1 - num_of_unks_target / tokens_target) * 100)
-        logger.info("Total: %d samples in %d buckets", sum(len(b) for b in self.data_source), len(self.buckets))
-        nsamples = 0
-        for bkt, buck, batch_size_seq, average_seq_len in zip(self.buckets,
-                                                              self.data_source,
-                                                              (bbs.batch_size for bbs in self.bucket_batch_sizes),
-                                                              self.data_target_average_len):
-            logger.info("Bucket of %s : %d samples in %d batches of %d, approx %0.1f words/batch",
-                        bkt,
-                        len(buck),
-                        math.ceil(len(buck) / batch_size_seq),
-                        batch_size_seq,
-                        batch_size_seq * average_seq_len)
-            nsamples += len(buck)
-        check_condition(nsamples > 0, "0 data points available in the data iterator. "
-                                      "%d data points have been discarded because they "
-                                      "didn't fit into any bucket. Consider increasing "
-                                      "--max-seq-len to fit your data." % ndiscard)
-        logger.info("%d sentence pairs out of buckets", ndiscard)
-        logger.info("fill up mode: %s", self.fill_up)
-        logger.info("")
-
-    def _populate_bucket_batch_sizes(self):
-        """
-        Compute bucket-specific batch sizes (sentences, average_words) and default bucket batch
-        size.
+class ShardedParallelSampleIter(BaseParallelSampleIter):
+    """
+    Goes through the data one shard at a time.
```

Comment:
Could mention that the order of shards is shuffled at each reset.

Reply:
done